sapl/protocoloadm/views.py

```diff
@@ -1600,3 +1604,41 @@ def filtra_tramitacao_destino_and_status(self, status, destino):
             status=status,
             unidade_tramitacao_destino=destino).distinct().values_list(
                 'documento_id', flat=True)
+
+
+def apaga_protocolos(user, ano):
+    all_protocolos = Protocolo.objects.filter(ano__in=ano)
+
+    for doc in DocumentoAdministrativo.objects.filter(protocolo__in=all_protocolos):
+        doc.protocolo = None
+        doc.save()
+
+    for ml in MateriaLegislativa.objects.filter(ano__in=ano):
+        ml.numero_protocolo = None
+        ml.save()
+
+    info = [str(p.numero)+'/'+str(p.ano) for p in all_protocolos]
+
+    auditoria = AuditoriaProtocolo(usuario=user,info=info)
+    auditoria.save()
+    all_protocolos.delete()
+
+@staff_member_required
+def apaga_protocolos_view(request):
+    if request.method == "GET":
+        if Protocolo.objects.exists():
+            ano_minimo = Protocolo.objects.all().order_by('ano').first().ano
+            intervalo_data = range(ano_minimo, datetime.now().year)
+        else:
+            intervalo_data  =None
+        return render(request,"protocoloadm/deleta_todos_protocolos.html",{'intervalo_data':intervalo_data})
+    
+    elif request.method == "POST":
+        password = request.POST.get('senha')
+        ano = list(map(int,request.POST.getlist('ano')))
```

Comment:
`anos = [int(i) for i in request.POST.getlist('ano')]`

Reply:
Resolvido.